sardana_albaem/ctrl/Albaem2OneDCtrl.py

```diff
@@ -0,0 +1,486 @@
+            Type: bool,
+            Description: 'Channel Digital inversion',
+            Memorize: NotMemorized,
+            Access: DataAccess.ReadWrite,
+            FGet: "get_Inversion",
+            FSet: "set_Inversion",
+
+        },
+        "InstantCurrent": {
+            Type: float,
+            Description: 'Channel instant current',
+            Memorize: NotMemorized,
+            Access: DataAccess.ReadOnly,
+            FGet: "get_InstantCurrent",
+        },
+        "FORMULA": {
+            Type: str,
+            Description: 'The formula to get the real value.\n '
+                            'e.g. "(value/10)*1e-06"',
+            Access: DataAccess.ReadWrite
+        },
+    }
+
+    @handle_error(msg="__init__: Could not connect to the device!")
+    def __init__(self, inst, props, *args, **kwargs):
+        """Class initialization."""
+        OneDController.__init__(self, inst, props, *args, **kwargs)
+        self.ip_config = (self.AlbaEmHost, self.Port)
+        self.albaem_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.albaem_socket.settimeout(1)
+        self.albaem_socket.connect(self.ip_config)
+        self.itime = 0.0
+        self.master = None
+        self._latency_time = 0.001  # In fact, it is just 320us
+        self._repetitions = 0
+        self.formulas = {1: 'value', 2: 'value', 3: 'value', 4:'value'}
+
+        self._points_per_step = 1
+
+        self.lock = Lock()
+
+    @debug_it
+    def AddDevice(self, axis):
+        """Add device to controller."""
+        pass
+
+    @debug_it
+    def DeleteDevice(self, axis):
+        """Delete device from the controller."""
+        # self.albaem_socket.close()
+        pass
+
+    @debug_it
+    def PrepareOne(self, axis, value, repetitions, latency, nb_starts):
+        self._is_aborted = False
+
+    
+    @debug_it
+    def PreStateAll(self):
+        pass
+
+    @debug_it
+    def StateAll(self):
+        """Read state of all axis."""
+        state = self.sendCmd('ACQU:STAT?')
+
+        if state in ['STATE_ACQUIRING', 'STATE_RUNNING']:
+            self.state = State.Moving
+
+        elif state == 'STATE_ON':
+            self.state = State.On
+
+        elif state == 'STATE_FAULT':
+            self.state = State.Fault
+
+        else:
+            self.state = State.Fault
+        self.status = state
+
+    @debug_it
+    def StateOne(self, axis):
+        """Read state of one axis."""
+        return self.state, self.status
+
+    @debug_it
+    @handle_error(msg="LoadOne: Could not configure the device!")
+    def LoadOne(self, axis, value, repetitions, latency_time):
+        if axis != 1:
+            raise Exception('The master channel should be the axis 1')
+
+        self.itime = value
+
+        # Set Integration time in ms
+        val = self.itime * 1000
+        if val < 0.1:   # minimum integration time 
+            self._log.debug("The minimum integration time is 0.1 ms")
+            val = 0.1
```

Comment:
is there any reason for not raising an exception instead? You are correcting the user silently

Reply:
yeah could be, again basing it on the ct controller, thinking should not diverge much.